op-service/eth/transactions.go

```diff
@@ -115,3 +115,28 @@ func CheckRecentTxs(
 	}
 	return oldestBlock.Uint64(), false, nil
 }
+
+type GenericTx interface {
+	// Transaction tries to interpret into a typed tx.
+	// This will return types.ErrTxTypeNotSupported if the transaction-type is not supported.
+	Transaction() (*types.Transaction, error)
+
+	// TxType returns the EIP-2718 type.
+	TxType() uint8
+
+	// TxHash returns the transaction hash.
+	TxHash() common.Hash
```

Comment:
@protolambda was it intentional to use `TxHash() common.Hash` rather than `Hash() common.Hash`? The latter might avoid some type conversions since `Hash() common.Hash` is an exisitng method on many types including the geth `types.Transaction`.

Reply:
`Hash()` can be confusing, since it's as generic as it can get. If you think `Hash()` is better than we can use that.